src/Microsoft.TestPlatform.Client/DesignMode/DesignModeClient.cs

```diff
@@ -347,7 +350,24 @@ public bool AttachDebuggerToProcess(int pid, CancellationToken cancellationToken
                 waitHandle.Set();
             };
 
-            _communicationManager.SendMessage(MessageType.EditorAttachDebugger, pid, _protocolConfig.Version);
+            // TODO: formalize this so we can the deprecation version from the message data, and automatically switch
```

Comment:
I am imagining a list of objects that hold 
- the message name, and are generically typed to the data that the message hold, so we can use that in calls to deserialize / serialize rather than always remember what message pair with what type. 
- Additional info about the message version and deprecation, so we can switch on it and send back the correct version of the message without harcoding  a magic number in place or a constant that holds that number. 

Something like:

```csharp 
new Payload<EditorAttachDebuggerPayload> { 
  MessageType = MessageType.EditorAttachDebugger2, 
  IntoducedInVersion = 7, 
  ReplacesMessage = MessageType.EditorAttachDebugger,
}
```